src/cascadia/TerminalApp/App.cpp

```diff
@@ -76,110 +77,43 @@ namespace winrt::TerminalApp::implementation
 
     // Method Description:
     // - Create all of the initial UI elements of the Terminal app.
-    //    * Creates the tab bar, initially hidden.
-    //    * Creates the tab content area, which is where we'll display the tabs/panes.
     //    * Initializes the first terminal control, using the default profile,
     //      and adds it to our list of tabs.
     void App::_Create(uint64_t parentHwnd)
     {
-        _tabView = MUX::Controls::TabView{};
-
-        _tabView.SelectionChanged({ this, &App::_OnTabSelectionChanged });
-        _tabView.TabClosing({ this, &App::_OnTabClosing });
-        _tabView.Items().VectorChanged({ this, &App::_OnTabItemsChanged });
-
-        _root = Controls::Grid{};
-
-        _tabRow = Controls::Grid{};
-        _tabRow.Name(L"Tab Row");
-        _tabContent = Controls::Grid{};
-        _tabContent.Name(L"Tab Content");
-
-        // Set up two columns in the tabs row - one for the tabs themselves, and
-        // another for the settings button.
-        auto tabsColDef = Controls::ColumnDefinition();
-        auto newTabBtnColDef = Controls::ColumnDefinition();
-        newTabBtnColDef.Width(GridLengthHelper::Auto());
-
-        _tabRow.ColumnDefinitions().Append(tabsColDef);
-        _tabRow.ColumnDefinitions().Append(newTabBtnColDef);
-
-        // Set up two rows - one for the tabs, the other for the tab content,
-        // the terminal panes.
-        auto tabBarRowDef = Controls::RowDefinition();
-        tabBarRowDef.Height(GridLengthHelper::Auto());
-        _root.RowDefinitions().Append(tabBarRowDef);
-        _root.RowDefinitions().Append(Controls::RowDefinition{});
-
-        _root.Children().Append(_tabRow);
-
-        Controls::Grid::SetRow(_tabRow, 0);
-
-        _root.Children().Append(_tabContent);
-        Controls::Grid::SetRow(_tabContent, 1);
-        Controls::Grid::SetColumn(_tabView, 0);
-
-        // Create the new tab button.
-        _newTabButton = Controls::SplitButton{};
-        Controls::SymbolIcon newTabIco{};
-        newTabIco.Symbol(Controls::Symbol::Add);
-        _newTabButton.Content(newTabIco);
-        Controls::Grid::SetRow(_newTabButton, 0);
-        Controls::Grid::SetColumn(_newTabButton, 1);
-        _newTabButton.VerticalAlignment(VerticalAlignment::Stretch);
-        _newTabButton.HorizontalAlignment(HorizontalAlignment::Left);
+        /* !!! TODO
+           This is not the correct way to host a XAML page. This exists today because we valued
+           getting a .xaml over tearing out all of the terminal logic and splitting it across App
+           and Page.
+           The work to clarify the boundary between app global state and "terminal page" state
+           is tracked in GH#1878.
+        */
+        auto terminalPage = winrt::make_self<TerminalPage>();
+        _root = terminalPage.as<winrt::Windows::UI::Xaml::Controls::Control>();
+        _tabContent = terminalPage->TabContent();

```

Comment:
Where did these methods (`TerminalPage::TabContent`, `TerminalPage::NewTabButton` etc.) come from? they're not defined in the idl, h, or cpp, so are they just xaml _magic_?

Reply:
Yep, they're autogenerated into `TerminalPageT` in `TerminalPage.xaml.g.hpp`